rn/Teacher/ios/Teacher/Submissions/QuizSubmissionList/View/QuizSubmissionListItemView.swift

```diff
@@ -0,0 +1,82 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Core
+import SwiftUI
+
+public struct QuizSubmissionListItemView: View {
+    private var model: QuizSubmissionListItemViewModel
+    private var action: () -> Void
+
+    public init(model: QuizSubmissionListItemViewModel, action: @escaping () -> Void) {
+        self.model = model
+        self.action = action
+    }
+
+    public var body: some View {
+        Button {
+            action()
+        } label: {
+            cellContent
+        }
+        .buttonStyle(PlainButtonStyle())
+        .accessibilityLabel(model.a11yLabel)
+    }
+
+    private var cellContent: some View {
+        HStack(alignment: .center, spacing: 14) {
+            Avatar(name: model.name, url: model.profileImageURL)
+                .frame(width: 36, height: 36)
+                .padding(.top, 5)
+            VStack(alignment: .leading, spacing: 2) {
+                Text(model.displayName)
+                    .font(.semibold16)
+                    .foregroundColor(.textDarkest)
+                    .lineLimit(1)
+                Text(model.status)
+                    .foregroundColor(model.statusColor)
+                    .font(.regular12)
+            }
+            Spacer()
```

Comment:
Use a spacer with 0 minimum value so in case of very compact layouts it won't consume precious space.
```suggestion
            Spacer(minLength: 0)
```